Sources/PackageLoading/ManifestJSONParser.swift

```diff
@@ -415,6 +411,33 @@ enum ManifestJSONParser {
         )
     }
 
+    /// Parses the URL type of a git repository
+    /// e.g. https://github.com/apple/swift returns "https"
+    /// e.g. git@github.com:apple/swift returns "git"
+    ///
+    /// This is *not* a generic URI scheme parser!
+    private static func parseScheme(_ location: String) -> String? {
```

Comment:
we can potentially also use Foundation API here, although not obvious which. for now migrated the code as is to preserve backwards compatibility. open to other ideas.

Reply:
Probably Foundation.URLComponents could help.  It does fall down pretty badly on the second one though.  So maybe try URLComponents(...).scheme first and then special-case `git@` prefix if that fails?  Not sure how much that actually buys.

Reply:
tried URLComponents but it does not handle the git@ prefix. @millenomi ?